oaiharvest/metadata.py

```diff
@@ -45,9 +45,10 @@ def writeMetadata(self, metadata_prefix, element, metadata):
 class XMLMetadataReader(object):
     """Really simple MetadataReader to serialize metadata to pretty XML."""
     def __call__(self, metadata_element):
-        return '\n'.join([tostring(rec_element,
-                                   method="xml",
-                                   pretty_print=True
-                                   )
-                          for rec_element
-                          in metadata_element])
+        # Six call fixes 'sequence item 0: expected str instance, bytes found'
+        return ('\n' if six.PY2 else b'\n').join(
```

Comment:
That solution produces
```
ERROR    string argument without an encoding
```
for me in py3 while working in python 2.

But I've found another solution involving calling `six.text_type()` on the result of `tostring()`. This allows me to unconditionally open the XML file in mode `w` instead of `wb` in Python3.